rest/handler.go

```diff
@@ -209,17 +213,24 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		if dbContext, err = h.server.GetDatabase(dbname); err != nil {
 			base.Infof(base.KeyHTTP, "Error trying to get db %s: %v", base.MD(dbname), err)
 
-			if shouldCheckAdminAuth {
-				if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+			if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+				if shouldCheckAdminAuth {
 					authorized, err := h.checkAdminAuthenticationOnly()
 					if err != nil {
 						return err
 					}
 
-					if authorized {
-						return base.HTTPErrorf(http.StatusForbidden, "")
+					if !authorized || h.providedAuthCredentials() {
+						return ErrInvalidLogin
+					}
```

Comment:
Line 225 looks like it should have `else { return base.HTTPErrorf(http.StatusForbidden, "")}` to handle the case of a an admin user that successfully authenticated requesting a db that doesn't exist.

Reply:
Fixed.